lib/help_scout/api/access_token.rb

```diff
@@ -9,11 +9,10 @@ def create
           response = connection.post('oauth2/token', token_request_params)
 
           case response.status
+          when 200 then new HelpScout::Response.new(response).body
```

Comment:
One difference in the two calls is the exception handling. In the access token request, I was going to make the response acceptance much more strict and only accept 200s. So in this case, I'm not sure we want to share exception handling. 

```rb?
case response.status
when 200 then new HelpScout::Response.new(response).body
when 429 then raise ThrottleLimitReached, response.body&.dig('error')
else raise InternalError, response.body
end
```

Reply:
I could see going more this route. When asking for a token, it's "200 or fail" -- that makes sense to me.

If we go that route though, I'd probably just go with:

```ruby
if response.ok?
  new(HelpScout:::Response.new(response).body)
else
  raise InternalError, "unexpected response (status #{response.status})"
end
```

Reply:
I grabbed the message for InternalError, but I still think throwing a ThrottleLimitReach exception is helpful here.

Reply:
Yeah, I wasn't recommending you eliminate those exception. I was thinking that this would use the shared response handling, so any of those other exception would get handled upstream and you'd be guaranteed at success response by the time you teach this code. So then it's a matter of raising if non-200 if we wanted to guarantee that extra safety.

Reply:
Ah that makes sense, I'd prefer to defer that refactor though if that's okay.

Reply:
I’m totally fine deferring that for now. It’s not currently too much code and I don’t see it having high churn.